evm/src/cpu/columns.rs

```diff
@@ -0,0 +1,135 @@
+pub const IS_SELFBALANCE: usize = IS_CHAINID + 1;
+pub const IS_BASEFEE: usize = IS_SELFBALANCE + 1;
+pub const IS_POP: usize = IS_BASEFEE + 1;
+pub const IS_MLOAD: usize = IS_POP + 1;
+pub const IS_MSTORE: usize = IS_MLOAD + 1;
+pub const IS_MSTORE8: usize = IS_MSTORE + 1;
+pub const IS_SLOAD: usize = IS_MSTORE8 + 1;
+pub const IS_SSTORE: usize = IS_SLOAD + 1;
+pub const IS_JUMP: usize = IS_SSTORE + 1;
+pub const IS_JUMPI: usize = IS_JUMP + 1;
+pub const IS_PC: usize = IS_JUMPI + 1;
+pub const IS_MSIZE: usize = IS_PC + 1;
+pub const IS_GAS: usize = IS_MSIZE + 1;
+pub const IS_JUMPDEST: usize = IS_GAS + 1;
+// Find the number of to push by reading the bottom 5 bits of the opcode.
+pub const IS_PUSH: usize = IS_JUMPDEST + 1;
+// Find the stack offset to duplicate by reading the bottom 4 bits of the opcode.
+pub const IS_DUP: usize = IS_PUSH + 1;
+// Find the stack offset to swap with by reading the bottom 4 bits of the opcode.
+pub const IS_SWAP: usize = IS_DUP + 1;
+pub const IS_LOG0: usize = IS_SWAP + 1;
+pub const IS_LOG1: usize = IS_LOG0 + 1;
+pub const IS_LOG2: usize = IS_LOG1 + 1;
+pub const IS_LOG3: usize = IS_LOG2 + 1;
+pub const IS_LOG4: usize = IS_LOG3 + 1;
+pub const IS_CREATE: usize = IS_LOG4 + 1;
+pub const IS_CALL: usize = IS_CREATE + 1;
+pub const IS_CALLCODE: usize = IS_CALL + 1;
+pub const IS_RETURN: usize = IS_CALLCODE + 1;
+pub const IS_DELEGATECALL: usize = IS_RETURN + 1;
+pub const IS_CREATE2: usize = IS_DELEGATECALL + 1;
+pub const IS_STATICCALL: usize = IS_CREATE2 + 1;
+pub const IS_REVERT: usize = IS_STATICCALL + 1;
+pub const IS_SELFDESTRUCT: usize = IS_REVERT + 1;
+
+pub const IS_INVALID_0: usize = IS_SELFDESTRUCT + 1;
+pub const IS_INVALID_1: usize = IS_INVALID_0 + 1;
+pub const IS_INVALID_2: usize = IS_INVALID_1 + 1;
+pub const IS_INVALID_3: usize = IS_INVALID_2 + 1;
+pub const IS_INVALID_4: usize = IS_INVALID_3 + 1;
+pub const IS_INVALID_5: usize = IS_INVALID_4 + 1;
+pub const IS_INVALID_6: usize = IS_INVALID_5 + 1;
+pub const IS_INVALID_7: usize = IS_INVALID_6 + 1;
+pub const IS_INVALID_8: usize = IS_INVALID_7 + 1;
+pub const IS_INVALID_9: usize = IS_INVALID_8 + 1;
+pub const IS_INVALID_10: usize = IS_INVALID_9 + 1;
+pub const IS_INVALID_11: usize = IS_INVALID_10 + 1;
+pub const IS_INVALID_12: usize = IS_INVALID_11 + 1;
+pub const IS_INVALID_13: usize = IS_INVALID_12 + 1;
+pub const IS_INVALID_14: usize = IS_INVALID_13 + 1;
+pub const IS_INVALID_15: usize = IS_INVALID_14 + 1;
+pub const IS_INVALID_16: usize = IS_INVALID_15 + 1;
+pub const IS_INVALID_17: usize = IS_INVALID_16 + 1;
+pub const IS_INVALID_18: usize = IS_INVALID_17 + 1;
+pub const IS_INVALID_19: usize = IS_INVALID_18 + 1;
+pub const IS_INVALID_20: usize = IS_INVALID_19 + 1;
+// An instruction is invalid if _any_ of the above flags is 1.
+
+pub const START_INSTRUCTION_FLAGS: usize = IS_STOP;
+pub const END_INSTRUCTION_FLAGS: usize = IS_INVALID_20 + 1;
+
+// If CPU cycle: the opcode, broken up into bits.
+// **big-endian** order
```

Comment:
FWIW Plonky2 uses little-endian elsewhere, for bignums and bitwise stuff (like `split_le`). Not a big deal if you think it's nicer / more standard in this context though.

Reply:
Yeah, I usually prefer little-endian, but I thought big-endian was nicer here. (Also, the EVM is big-endian.)